app/src/main/res/layout/activity_product_detail.xml

```diff
@@ -1,85 +1,109 @@
 <?xml version="1.0" encoding="utf-8"?>
-<ScrollView
+<android.support.constraint.ConstraintLayout
```

Comment:
I'm not sure why you've nested the ConstraintLayout's here, ideally we should keep our layout hierarchy as flat as possible and I don't see why this layout couldn't be flat with the visibility toggled on the views that aren't showing? Come grab me if not clear.